src/components/application_manager/src/rpc_passing_handler.cc

```diff
@@ -93,6 +93,63 @@ bool RPCPassingHandler::CanHandleFunctionID(int32_t function_id) {
   return false;
 }
 
+bool RPCPassingHandler::IsPassthroughAllowed(
+    smart_objects::SmartObject rpc_message) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  mobile_api::FunctionID::eType function_id =
+      static_cast<mobile_api::FunctionID::eType>(
+          rpc_message[strings::params][strings::function_id].asInt());
+  uint32_t connection_key =
+      rpc_message[strings::params][strings::connection_key].asUInt();
+  std::string function_id_str =
+      MessageHelper::StringifiedFunctionID(function_id);
+  ApplicationSharedPtr app = app_manager_.application(connection_key);
+  PolicyHandlerInterface& policy_handler = app_manager_.GetPolicyHandler();
+  if (!app) {
+    return false;
+  } else if (function_id_str.empty()) {
+    // Unknown RPC, just do basic revoked and user consent checks
+    std::string device_mac;
+    app_manager_.connection_handler().get_session_observer().GetDataOnDeviceID(
+        app->device(), NULL, NULL, &device_mac, NULL);
+    return policy_handler.UnknownRPCPassthroughAllowed(app->policy_app_id()) &&
+           !policy_handler.IsApplicationRevoked(app->policy_app_id()) &&
+           policy::kDeviceAllowed ==
+               app_manager_.GetUserConsentForDevice(device_mac);
+  }
+
+  RPCParams params;
```

Comment:
@jacobkeeler 
can we extract RPC params checking to separate function to make this a bit more easy to read?